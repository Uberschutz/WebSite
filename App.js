/*
** EPITECH INNOVATIVE PROJECT, 2018
** Ãœberschutz
** File description:
** Backend
*/


var express = require('express');
var bodyParser = require('body-parser');
var path = require("path");
var app = express();

app.use(bodyParser.json()); // for parsing application/json

app.use(express.static(__dirname + '/template'));

app.get('/', function(req, res) {
	res.sendFile(path.join(__dirname+'/index.html'));
});

var port = 7171;
const hostname = '127.0.0.1';
var server = app.listen(port, function() {
	console.log('Server running at http://' + hostname + ':' + port + '/');
	const all_routes = require('express-list-endpoints');
	console.log(all_routes(app));
});