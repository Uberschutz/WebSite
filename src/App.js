import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import './styles/App.css';

import HomePage from "./pages/HomePage";
import Header from "./pages/Header";
import Footer from "./pages/Footer";
import Connexion from "./pages/Connexion";
import {Contact} from "./pages/Contact";
import Profil from "./pages/Profil";
import Parametres from "./pages/Parametres";

class App extends Component {
  render() {
    return (
        <div>
          <Router>
          <Header/>
          <Switch>
            <Route path={'/'} exact strict component={HomePage}/>
            <Route path={'/Connexion'} component={Connexion}/>
            <Route path={'/Contact+FAQ'} component={Contact}/>
            <Route path={'/Profil'} component={Profil}/>
            <Route path={'/Parametres'} component={Parametres}/>
          </Switch>
          <Footer/>
          </Router>
        </div>
    );
  }
}

export default App;
