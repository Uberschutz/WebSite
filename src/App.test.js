import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {BrowserRouter as Router} from "react-router-dom";

it('renders whole without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
