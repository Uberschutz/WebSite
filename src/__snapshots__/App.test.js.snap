// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the whole app rendering renders whole application without crashing 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <nav
        class="navbar navbar-expand-sm uber-color navbar navbar-expand-md navbar-light"
      >
        <img
          alt="logo"
          height="70"
          src="Uberschutz-Icon.png"
          width="65"
        />
        <a
          class="navbar-brand uber-color button-footer navbar-brand"
          href="/"
        >
           Überschutz
        </a>
        <div
          class="collapse navbar-collapse collapse show navbar-collapse"
        >
          <ul
            class="ml-auto navbar  navbar-expand-md navbar-nav"
          >
            <li
              class="nav-item nav-item"
            >
              <a
                class="nav-link uber-color"
                href="/"
              >
                Menu
              </a>
            </li>
            <li
              class="nav-item nav-item"
            >
              <a
                class="nav-link uber-color"
                href="/Connection"
              >
                Connexion
              </a>
            </li>
            <li
              class="nav-item nav-item"
            >
              <a
                class="nav-link uber-color"
                href="/Contact+FAQ"
              >
                Contact + FAQ
              </a>
            </li>
            <li
              class="nav-item nav-item"
            >
              <a
                class="nav-link uber-color"
                href="/Subscription"
              >
                Abonnements
              </a>
            </li>
          </ul>
        </div>
        <div
          class="btn-group btn-group-sm"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-primary"
            type="button"
          >
            <img
              alt="french"
              class="clickable-flag touched"
              height="35"
              src="icons8-france-96.png"
              width="35"
            />
          </button>
          <div
            aria-hidden="true"
            class="drop btn dropdown-menu"
            role="menu"
            tabindex="-1"
          >
            <div>
              <img
                alt="french"
                class="clickable-flag touched"
                data-lang="fr"
                height="35"
                src="icons8-france-96.png"
                width="35"
              />
            </div>
            <div>
              <img
                alt="english"
                class="clickable-flag"
                data-lang="en"
                height="33"
                src="icons8-grande-bretagne-48.png"
                width="35"
              />
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div>
      <img
        alt="canvas"
        class="responsive-image"
        src="canvas2.png"
      />
      <span
        class="text-span"
      >
        Internet :
        <br />
         
        êtes-vous conscients des dangers
        <br />
         
        que rencontrent vos enfants ?
      </span>
      <div
        class="row responsive-image Home"
      >
        <div>
          <br />
          <br />
          <img
            alt="whois"
            class="responsive-image"
            src="CacherPseudo.jpg"
          />
        </div>
        <div
          class="txt-intro responsive-image"
        >
          <h3
            class="h7-font"
          >
            Notre solution :
          </h3>
          <h6
            class="h6-font"
          >
            Pour une navigation saine, utilisez Überschutz.
            <br />
            Ce logiciel non bloquant vous préviendra lorsque votre enfant se met en danger sur Internet.
          </h6>
          <h8
            class="italic-txt"
          >
            Suicide, drogue, anorexie, harcèlement, contenus pornographiques ... Vous n'imaginez pas à quoi il est confronté.
          </h8>
        </div>
      </div>
      <div
        class="Home responsive-image"
      >
        <br />
        <br />
        <h5
          class="rappel-info h7-font"
        >
          En quelques mots :
        </h5>
        <br />
        <br />
        <div
          class="row responsive-image"
        >
          <div
            class="col"
          >
            <img
              alt="software"
              src="icons8-boite-à-logiciel-64.png"
            />
            <br />
            <br />
            <span
              class="rappel-info"
            >
              Un logiciel
            </span>
          </div>
          <div
            class="col responsive-image"
          >
            <img
              alt="navigation"
              src="icons8-carte-au-trésor-150.png"
            />
            <br />
            <br />
            <span
              class="rappel-info"
            >
              Non bloquant
            </span>
          </div>
          <div
            class="col responsive-image"
          >
            <img
              alt="blocking"
              src="icons8-cadenas-100.png"
            />
            <br />
             
            <br />
            <span
              class="rappel-info"
            >
              Non intrusif
            </span>
          </div>
        </div>
        <br />
        <div
          class=""
        >
          <h7
            class="rappel-info responsive-image"
          >
            Pour protéger votre enfant et discuter avec lui lorsqu'une situation compliquée se présente.
            <br />
          </h7>
          <br />
        </div>
      </div>
    </div>
    <footer
      class="uber-color2"
    >
      <div>
        <a
          href="/Contact+FAQ"
        >
          <button
            class="btn btn-primary button-footer"
            type="button"
          >
            Contact
          </button>
        </a>
        <br />
        <span>
          © 2018 - 
          2020
        </span>
        <br />
        <span>
          Créé par Marianne & Thomas 
          <br />
          Icônes de l'application par 
          <a
            href="https://icones8.fr"
          >
            icons8
          </a>
           & 
          <a
            href="https://www.flaticon.com/authors/dmitri13"
            title="dmitri13"
          >
            dmitri13
          </a>
          <br />
        </span>
      </div>
      <div
        class="button-footer row txt-align"
      >
        <a
          href="/DataInformations"
        >
          <span
            class="col"
          >
            Données personnelles
          </span>
        </a>
        <a
          href="/Others"
        >
          <span
            class="col"
          >
            Nos autres services
          </span>
        </a>
      </div>
    </footer>
  </div>
</div>
`;
